app/models/service_orchestration.rb

```diff
@@ -100,7 +100,7 @@ def all_vms
   end
 
   def add_resource(rsc, _options = {})
-    raise "Service Orchestration subclass does not support add_resource for #{rsc.class.name}" unless rsc.kind_of?(OrchestrationStack)
+    raise "Service Orchestration subclass does not support add_resource for #{rsc.class.name}" unless rsc.try(:valid_service_orchestration_resource)
```

Comment:
Can we set the base class to be `false` and have these 3 classes override with `true` ?

Reply:
I like the removal of the hardcoded class name 👍